### PR TITLE
Add rpcserver test for `RenewAccount` endpoint

### PR DESCRIPTION
**Pull request by positiveblue**
Code extracted from #323 (removing the sponsoring part)

### CODE REVIEW

File: rpcserver_test.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	return req
+}
+
+var renewAccountTestCases = []struct {
+	name          string
+	getReq        func() *poolrpc.RenewAccountRequest
+	checkResponse func(*poolrpc.RenewAccountResponse) error
+	mockSetter    func(*poolrpc.RenewAccountRequest,
+		*account.MockManager, *MockMarshaler,
+	)
```
Comment (guggero) [inline rpcserver_test.go:57]: nit: closing parenthesis of method definition should be on same line as last argument (to better distinguish from method call). Same in the actual implementation below.

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

File: order/interfaces.go [hunk 2]
```diff
@@ -23,6 +23,13 @@ import (
 // Nonce is a 32 byte pseudo randomly generated unique order ID.
 type Nonce [32]byte
 
+// The size of a SHA256 checksum in bytes.
+//
+// Note: this matches the sha256.Size definition. However, mockgen
+// complains about not being able to find the sha256 package. When
+// that bug is fixed, we can change back to [sha256.Size]byte.
+const size = 32
```
Comment (bhandras) [inline order/interfaces.go:31]: nit: `checksumSize`/`hashSize`?

File: account/manager.go [hunk 3]
```diff
@@ -123,7 +123,7 @@ type ManagerConfig struct {
 }
 
 // Manager is responsible for the management of accounts on-chain.
-type Manager struct {
+type manager struct {
```
Comment (bhandras) [inline account/manager.go:126]: It's helpful to add a compile time enforcer for the interface:
```
var _ Manager = (*manager)(nil)
```

File: rpcserver_test.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+			srv.bestHeight = 100
+
+			ctx, cancel := context.WithTimeout(
+				context.Background(), ctxTimeout,
+			)
+			defer cancel()
+
+			resp, err := srv.RenewAccount(ctx, req)
+			if tc.expectedError != "" {
+				assert.EqualError(t, err, tc.expectedError)
```
Comment (bhandras) [inline rpcserver_test.go:168]: nit: `require` and the expectation comes first iirc

**Review by bhandras — APPROVED** [review]
LGTM :hamburger:

File: rpcserver_test.go (hunk 4 above)
Comment (positiveblue) [reply rpcserver_test.go:163]: It looks like for `EqualError` this is the right order? 

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1367

**Pull request merged**